### DISCUSSION TITLE
Is there a way to block copy/paste from within SEB

### DISCUSSION THREAD

**Discussion by dhandby** [SafeExamBrowser/seb-win-refactoring#118] [Q&A]
SEB windows 3.2.0 and SEB macosx 2.3.2

Is there a way in order to restrict copy/paste action within SEB itself?

We know about SEB blocking this ability from outside SEB and coming into the Exam with data which is blocked.

Use case.
We would like to whitelist a website within SEB, but want to deny copy/paste from that into the quiz.

Tests performed.
1. Currently when whitelisting a website you can copy / paste by default anything.
2. When enabling "Full screen popups with Javascript security" in the quiz setting. This is suppose to block data being copied into this field but it still allows the data to be copied. 
A prompt is displayed
![image](https://user-images.githubusercontent.com/26612081/125889881-14aa396a-6449-4003-b168-80fda5660b3c.png)

Use private clipboard: When enabled, cut/copy/paste works only with content inside the SEB browser, the private clipboard doesn't accept content from other applications or tools. This prevents utilities running in the background and cloud clipboard features to be used to copy-paste contents into or out of exams. Private clipboard should always be used besides when working with third party applications in a managed user account.

**Comment by dbuechel** [comment]
No, SEB 3.x currently doesn't have a private clipboard functionality. Furthermore, having a private clipboard for individual web pages within the browser itself is something that we probably won't add, as it seems to be too cumbersome to implement on first glance. If you have control over the other, whitelisted webpage, then you should be able to suppress copy & paste via JavaScript code though.

**Reply by dhandby** [reply]
Hi, Sorry for this. I believe I posted the MAC side of 2.3.2 which actually copies the content across with the above photo and windows 3.2.0 successfully blocks when entering in the copied data.

**Reply by 1TheITCrowd** [reply]
Would it be possible to have an option to fully suppress copy/paste? I'm not talking about a clipboard for individual pages but really prevent users from using the clipboard. Not even a private Clipboard. Please contact me for a private conversation.

Is an Issue Report public immediately or can I send you private information somehow?

**Reply by dbuechel** [reply]
@1TheITCrowd Please feel free to create a feature request as an issue, suppressing CTRL+C and CTRL+V definitely should be possible.
